justfile

```diff
@@ -11,12 +11,28 @@ check:
 
 # Lint everything.
 lint:
-  cargo clippy --workspace --all-targets --all-features -- --deny warnings
+  cargo +nightly clippy --workspace --all-targets --all-features -- --deny warnings
 
 # Check the formatting
 format:
   cargo +nightly fmt --all --check
 
+# Quick and dirty CI useful for pre-push checks.
+sane:
+  cargo test --workspace --all-targets --no-default-features
+  cargo test --workspace --all-targets
+  cargo test --workspace --all-targets --all-features
+
+  cargo test --manifest-path bitcoin/Cargo.toml --doc --all-features
+
+  # Make an attempt to catch feature gate problems in doctests
+  cargo test --manifest-path bitcoin/Cargo.toml --doc --no-default-features
+  cargo test --manifest-path units/Cargo.toml --doc --no-default-features
+  cargo test --manifest-path io/Cargo.toml --doc --no-default-features
+  cargo test --manifest-path hashes/Cargo.toml --doc --no-default-features
+
+  just lint
```

Comment:
Is it a requirement to launch this at the end?

If it isn't it's probably more conventional to write "lint" after "sane:"
Which like in the makefile means launch this jobs before this one

Reply:
It looks like to call `lint` from `sane` you must call `just lint` not `lint`..

Reply:
yes of course, but if it's fine to execute the `lint` job before the commands in the `sane` job, you can write:

```
sane: lint
   cargo test --workspace --all-targets --no-default-features
   ....
```
and in this case is the lint job is called as a prerequisite of the sane job

(Note `lint` is on the same line of `sane:`)

Reply:
example

```shell
$ cat justfile 
default:
    echo 'Hello, world!'

job: default
    echo "Banana"
$ just job
echo 'Hello, world!'
Hello, world!
echo "Banana"
Banana

```

Reply:
interesting that

```
sane: lint
   cargo test --workspace --all-targets --no-default-features
```

works but 
```
sane: 
   lint
   cargo test --workspace --all-targets --no-default-features
```

does not.  

I don't think it matters if the lint is run before or after sane in this case.

Reply:
This is a mute point, new function `just sane` no longer runs the linter.

Reply:
Why? Seems like linting would be useful to do before CI.

Reply:
I was incorrect in my understanding how different toolchain builds interacted. FTR if one runs `clippy` with `nightly` followed by building with `stable`, then `clippy +nightly` again, no rebuild is required.

I've added linting back in as part of the job and also quietened down the output.